main/externalData/index.js

```diff
@@ -216,9 +214,9 @@ function stop () {
   log.info('stopping external data worker')
 
   currentNetworkObserver.remove()
-  allNetworksObserver.remove();
+  allNetworksObserver.remove()
 
-  [heartbeat, allAddressScan, trackedAddressScan, nativeCurrencyScan, inventoryScan]
+  ;[heartbeat, allAddressScan, trackedAddressScan, nativeCurrencyScan, inventoryScan]
```

Comment:
;

Reply:
this is intentional, you have to insert a semicolon before starting a line with a bracket. in the old code I had it at the end of the previous line but if you have to make edits that can be error prone, this is the accepted best practice.

the reason is that the bracket is also the indexing syntax so having the previous line be `allNetworksObserver.remove()[ ... ]` would also be syntactically correct and this is how it gets interpreted

Reply:
could change it to `new Array(heartbeat, allAddressScan, trackedAddressScan, nativeCurrencyScan, inventoryScan)
    .forEach(scanner => { if (scanner) clearInterval(scanner) })`

if you think that's easier to read

Reply:
Yeah, I would normally follow something like:
```
const scanners = [heartbeat, allAddressScan, trackedAddressScan, nativeCurrencyScan, inventoryScan]
scanners.forEach(scanner => { if (scanner) clearInterval(scanner) })
```
but `new Array` works for me too.

I think avoiding the use of `;` this way is easier to maintain in the long run.

Reply:
👍